discovery/eureka.go

```diff
@@ -39,11 +39,17 @@ func NewEurekaClient(eurekaUrl string) (ec EurekaClient, err error) {
 
 var errNoIpsFound = errors.New("No IPs associated to the requested App name")
 
+// TODO: Probably we should break this function
 func (ec EurekaClient) GetIPs(appName string) ([]string, error) {
 	eurekaAppUrl := ec.eurekaUrl + "/v2/apps/" + appName
-	//resp, err := http.Get(eurekaAppUrl, "application/json; charset=utf-8")
+	// can we reuse the client but starting from 0 in terms of timeout? to review
 	httpclient := http.Client{Timeout: time.Second * eurekaClientTimeoutInSeconds}
-	resp, err := httpclient.Get(eurekaAppUrl)
+	req, err := http.NewRequest("GET", eurekaAppUrl, nil)
```

Comment:
why did you change from httpclient.Get to http.NewRequest + httpclient.Do?

Reply:
Because I'm trying to add a request header, and the Get method does not allow me so